scipy/optimize/lbfgsb.py

```diff
@@ -268,42 +269,80 @@ def _minimize_lbfgsb(fun, x0, args=(), jac=None, bounds=None,
     relationship between the two is ``ftol = factr * numpy.finfo(float).eps``.
     I.e., `factr` multiplies the default machine floating-point precision to
     arrive at `ftol`.
-
+    If any of the lower bounds are equal to the upper bounds then the
+    corresponding entry in ``OptimizeResult.jac`` will be np.nan.
```

Comment:
I think `nan` is the appropriate choice. After all, I think that the main thing we didn't like about the other approach is that there was no basis for setting the derivatives to zero.